### PR TITLE
Dragonflight Patch V5

### PR DESCRIPTION
**Pull request by MleveK**
- Fixed Container issues
- Fixed Tooltip Processor
- Fixed GetContainerItemInfo
- Fixed ClassID and SubClassID Values
- Fixed Item Count
- Fixed Item Borders
- Fixed Filtering
- Fixed Icons (Textures)
- Added Reagent Bag
- Added Separate Free Space Count for Reagent Bag

### CODE REVIEW

File: modules/NewItemTracking.lua [hunk 1]
```diff
@@ -243,7 +245,11 @@ end
 function mod:ShowBlizzardGlow(button, enable)
 	if not button.NewItemTexture then return end
 	if enable then
-		local _, _, _, quality = GetContainerItemInfo(button.bag, button.slot)
+		local itemInfo = GetContainerItemInfo(bag, slot)
```
Comment (juntereiner) [inline modules/NewItemTracking.lua:248]: I had some errors, it's button.bag and button.slot

File: modules/NewItemTracking.lua [hunk 2]
```diff
@@ -243,7 +245,11 @@ end
...
+		local quality
+	if itemInfo ~= nil then
+		local quality = itemInfo.quality
```
Comment (juntereiner) [inline modules/NewItemTracking.lua:251]: don't need local

**Comment by juntereiner** [comment]
Thank you for your work,, it's working great now :)

I see you added a reagent bag for the bank too, but in my knowledge there isn't any. It's only for your bags.

File: widgets/ContainerFrame.lua [hunk 3]
```diff
@@ -769,6 +793,10 @@ local function FilterByBag(slotData)
 		name = REAGENT_BANK
 	elseif bag <= NUM_BAG_SLOTS then
 		name = format(L["Bag #%d"], bag)
+	elseif addon.isRetail then
+		if bag == REAGENTBAG then
+			name = format(L["Reagent Bag"])
+		end
```
Comment (juntereiner) [inline widgets/ContainerFrame.lua:799]: here name could be nil and it produces an error with the strjoin below. add name = format(L["Bank bag #%d"], bag - NUM_BAG_SLOTS)

**Comment by MleveK** [comment]
Huge Thanks @juntereiner for all of your feedback\debugging

**Comment by cindynico** [comment]
i see there is a fix for the reagent bag but i dont see any updates in curse, is there a seperate file somewhere here?

**Comment by MleveK** [comment]
@cindynico
It's not on Curse as this is has not been approved by the Addon Owner yet. 
You can manually download it here for now.

[AdiBagsV5.1.zip](https://github.com/AdiAddons/AdiBags/files/10045165/AdiBagsV5.1.zip)

**Comment by cindynico** [comment]
unzipped and pasted it to my addons. Didnt see any reagent bag when I logged in but I geuss thats merged with the other empty slots and something visual like a categorie. Which is great work if it uses it auto. In anyway, thank you for the great work. I know people can be upset if something doesnt work but these problems arent something you can fix before a launch. Anyway just wanted to get that out there and again thank you all for the effort!

**Comment by Trantore** [comment]
Amazing work MieveK - only thought about coming on GitHub this morning - have been suffering with the Blizzard bag view since maintenance.

AdiBags is one of my major 'quality of life' addons so glad to see it back working so quickly.

**Comment by juntereiner** [comment]
If anyone is interested, i've modified some adibags addons to work with current version and without creating frames.
It includes Archaelogy, Artifact weapons, battlepetitem, boe boa, fishing item, hearthstones, lowlevel, openable.
 
[adibags_addons_fix.zip](https://github.com/AdiAddons/AdiBags/files/10047129/adibags_addons_fix.zip)

**Comment by MleveK** [comment]
I'll keep putting there here as I know it's easier for most to get.

[AdiBagsV5.2.zip](https://github.com/AdiAddons/AdiBags/files/10047220/AdiBagsV5.2.zip)

**Comment by MleveK** [comment]
Sorry for spamming with so many files\fixes.
This is for Border Glow Fix [e2ae030](https://github.com/AdiAddons/AdiBags/pull/790/commits/e2ae0304b5127b8e42a8598f5158f758cffbc3e8)

[AdiBagsV5.3.zip](https://github.com/AdiAddons/AdiBags/files/10047563/AdiBagsV5.3.zip)

**Comment by juntereiner** [comment]
> Sorry for spamming with so many files\fixes. This is for Border Glow Fix [e2ae030](https://github.com/AdiAddons/AdiBags/pull/790/commits/e2ae0304b5127b8e42a8598f5158f758cffbc3e8)
> 
> [AdiBagsV5.3.zip](https://github.com/AdiAddons/AdiBags/files/10047563/AdiBagsV5.3.zip)

Is this the same thing as https://github.com/AdiAddons/AdiBags/pull/790#issuecomment-1320925739 ?

**Comment by MleveK** [comment]
> > Sorry for spamming with so many files\fixes. This is for Border Glow Fix [e2ae030](https://github.com/AdiAddons/AdiBags/pull/790/commits/e2ae0304b5127b8e42a8598f5158f758cffbc3e8)
> > [AdiBagsV5.3.zip](https://github.com/AdiAddons/AdiBags/files/10047563/AdiBagsV5.3.zip)
> 
> Is this the same thing as [https://github.com/AdiAddons/AdiBags/pull/790#issuecomment-1320925739 ?](https://github.com/AdiAddons/AdiBags/pull/759/commits/2514c36899c3bece9f9eb00f90c1a00ad03f7e07)

self.hasItem just gets reset to 1 for certain items for some reason, and Yeah that function wow what was causing them to not update their border. But I didn't change that particular function though, I wasn't sure if the hasItem() function had any other reason to have 1 as false. I instead just reinitialized self.hasItem to True or False before updating the function.

**Comment by glassleo** [comment]
Amazing work! Any chance you could get Masque working? 😄

**Comment by cindynico** [comment]
> I'll keep putting these here as I know it's easier for most to get. Notes are in the commit above [6409ae9](https://github.com/AdiAddons/AdiBags/pull/790/commits/6409ae9cf669f5daeefbd45acefc0adeb74a1452) @cindynico I separate the Reagent Bag image
> 
> Added Reagent Bag under Freespace with RB image
> 
> Upgrade Arrow is back, and moved to see. (Pawn Addon Required) image
> 
> [AdiBagsV5.2.zip](https://github.com/AdiAddons/AdiBags/files/10047220/AdiBagsV5.2.zip)

thank you vm. I am keeping this page open and come in to check once in a while :)

**Comment by set4812** [comment]
@MleveK you need rebase branch with master, and a good job.

**Comment by Cidan** [comment]
Thank you @MleveK for all your incredible work here -- I'm going to test this branch late tomorrow/early Monday as time permits and will try to push a release early next week. I want to review the code in-depth and actually give real feedback when I'm back.

**Comment by Cidan** [comment]
One thing that would be immensely helpful, @MleveK, is if you could provide a larger description around your design decisions and why you chose to write things the way you did, i.e. how did you take into account future maintenance, portability between the various WoW products, etc. These will be the key areas that I will be looking at, beyond a binary "does it work or not" evaluation.

Thanks!

**Comment by glassleo** [comment]
> > Sorry for spamming with so many files\fixes. This is for Border Glow Fix [e2ae030](https://github.com/AdiAddons/AdiBags/pull/790/commits/e2ae0304b5127b8e42a8598f5158f758cffbc3e8)
> > [AdiBagsV5.3.zip](https://github.com/AdiAddons/AdiBags/files/10047563/AdiBagsV5.3.zip)
> 
> Is this the same thing as [https://github.com/AdiAddons/AdiBags/pull/790#issuecomment-1320925739 ?](https://github.com/AdiAddons/AdiBags/pull/759/commits/2514c36899c3bece9f9eb00f90c1a00ad03f7e07)

I've been using this version for a while and I get a lua error whenever moving an item to the "free space" button (pardon me if you've already fixed this)

> Interface/AddOns/AdiBags/widgets/ItemButton.lua:500: Usage: GetItemInfo(itemID|"name"|"itemlink")

I was able to fix it by changing ``widgets\ItemButton.lua`` to guard for nil on ``self.itemLink and self.itemId):

```lua
local quality = hasItem(self.hasItem) and (self.itemLink or self.itemId) and select(3, GetItemInfo(self.itemLink or self.itemId))
```

**Comment by MalikiOne** [comment]
Getting this error when leaving auctionhouse - 

`7x AdiBags/widgets/ContainerFrame.lua:1117: Action[SetPoint] failed because[Cannot anchor to a region dependent on it]: attempted from: AdiBagsSection6:SetPoint.
[string "=[C]"]: in function `SetPoint'
[string "@AdiBags/widgets/ContainerFrame.lua"]:1117: in function `LayoutSections'
[string "@AdiBags/widgets/ContainerFrame.lua"]:1199: in function <AdiBags/widgets/ContainerFrame.lua:1167>
[string "@Ace3/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:124: in function <...Ons/Ace3/CallbackHandler-1.0/CallbackHandler-1.0.lua:124>
[string "=[C]"]: ?
[string "@Ace3/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:24: in function <...Ons/Ace3/CallbackHandler-1.0/CallbackHandler-1.0.lua:20>
[string "@Ace3/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:59: in function `SendMessage'
[string "@AdiBags/core/Core.lua"]:450: in function `?'
[string "@Ace3/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:114: in function <...Ons/Ace3/CallbackHandler-1.0/CallbackHandler-1.0.lua:114>
[string "=[C]"]: ?
[string "@Ace3/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:24: in function <...Ons/Ace3/CallbackHandler-1.0/CallbackHandler-1.0.lua:20>
[string "@Ace3/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:59: in function <...Ons/Ace3/CallbackHandler-1.0/CallbackHandler-1.0.lua:54>
[string "=(tail call)"]: ?
[string "=[C]"]: in function `CloseAuctionHouse'
[string "@Blizzard_AuctionHouseUI/Blizzard_AuctionHouseFrame.lua"]:488: in function <...izzard_AuctionHouseUI/Blizzard_AuctionHouseFrame.lua:478>
[string "=[C]"]: in function `Hide'
[string "@FrameXML/UIParent.lua"]:2841: in function `SetUIPanel'
[string "@FrameXML/UIParent.lua"]:2868: in function `MoveUIPanel'
[string "@FrameXML/UIParent.lua"]:2892: in function `HideUIPanel'
[string "@FrameXML/UIParent.lua"]:2565: in function <FrameXML/UIParent.lua:2557>
[string "=[C]"]: in function `SetAttribute'
[string "@FrameXML/UIParent.lua"]:3179: in function `HideUIPanel'
[string "@FrameXML/UIParent.lua"]:1994: in function <FrameXML/UIParent.lua:1362>`

**Comment by blehmonster** [comment]
Getting this error almost constantly when interacting with objects, etc.:

10x AdiBags/widgets/ItemButton.lua:500: Usage: GetItemInfo(itemID|"name"|"itemlink")
[string "=[C]"]: ?
[string "@AdiBags/widgets/ItemButton.lua"]:500: in function `UpdateBorder'
[string "@AdiBags/widgets/ItemButton.lua"]:294: in function `?'
[string "@LibSharedMedia-3.0-8020003/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:119: in function <...edia-3.0/CallbackHandler-1.0/CallbackHandler-1.0.lua:119>
[string "=[C]"]: ?
[string "@LibSharedMedia-3.0-8020003/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:29: in function <...edia-3.0/CallbackHandler-1.0/CallbackHandler-1.0.lua:25>
[string "@LibSharedMedia-3.0-8020003/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:64: in function <...edia-3.0/CallbackHandler-1.0/CallbackHandler-1.0.lua:59>
[string "=(tail call)"]: ?

**Comment by Meilinn** [comment]
@MleveK and @Cidan Thank you both for taking the time to fix this addon. It's the only addon I miss when patches broke them. I can play without dbm or healbot, but not without adibags. Cidan, I hope your work trip was good. Thank you all

**Comment by cindynico** [comment]
Hi @MleveK I am experiencing display freeze when opening bags. only for a second but it feels like my PC is getting stuck, annoying feeling. It happens every time, I open my bags which is often seeing I portal with the guild cloacks and am mount hunting with about 40 toons. I dont have an error lua seeing I disabled that so I am hoping you know what I am talking about and can help :) Maybe you have other people with similar experiences. Thank you in advance for looking into it.

**Comment by glassleo** [comment]
> Hi @MleveK I am experiencing display freeze when opening bags. only for a second but it feels like my PC is getting stuck, annoying feeling. It happens every time, I open my bags which is often seeing I portal with the guild cloacks and am mount hunting with about 40 toons. I dont have an error lua seeing I disabled that so I am hoping you know what I am talking about and can help :) Maybe you have other people with similar experiences. Thank you in advance for looking into it.

Is it still happening with all addons disabled except AdiBags? I'd guess it's another addon.

**Comment by juntereiner** [comment]
@cindynico You probably have some adibags plugin which freeze your game because they create a frame for each item in your bag. I had that too and I modified some of them to fix it. Look here: https://github.com/AdiAddons/AdiBags/issues/782#issuecomment-1320704382

**Comment by cindynico** [comment]
@MleveK I updated from curse yd and I see my seperate reagent bag isnt there anymore, wasnt that in the recent update and does this mean i have to install the update from above again? Also screen still freezes a few seconds everytime I open bags. With 40 toons thats a lot :D

**Comment by Cidan** [comment]
@Frurox I'm working on it as we speak.

Sorry for the delay everyone, my daughter got sick over the holiday weekend, and I was on dad duty. Trying to push a release with some fixes today.

**Comment by Nifelhein** [comment]
> @Frurox I'm working on it as we speak.
> 
> Sorry for the delay everyone, my daughter got sick over the holiday weekend, and I was on dad duty. Trying to push a release with some fixes today.

I sincerely only care that she is doing better, sick kids wreck our sleep and also crush our hearts.

**Comment by Frurox** [comment]
@Cidan and @Nifelhein Thank you for the update and hard work on this.

**Comment by Cidan** [comment]
Thanks folks, that means a lot!

A new release is going out right now, .49, which contains most of the fixes in this patch. I've made a few small, minor changes around how things were implemented, favoring @jnwhiteh's abstraction.

I'm going to leave this PR open through the next few days, then close it. If anything comes up, please open a new issue.

Thanks!

**Pull request closed without merging**